packages/frontend/core/src/components/affine/page-properties/table.tsx

```diff
@@ -0,0 +1,892 @@
+  Tooltip,
+} from '@affine/component';
+import { useCurrentWorkspacePropertiesAdapter } from '@affine/core/hooks/use-affine-adapter';
+import { useBlockSuitePageBacklinks } from '@affine/core/hooks/use-block-suite-page-backlinks';
+import type {
+  PageInfoCustomProperty,
+  PageInfoCustomPropertyMeta,
+} from '@affine/core/modules/workspace/properties/schema';
+import { timestampToLocalDate } from '@affine/core/utils';
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import { assertExists } from '@blocksuite/global/utils';
+import {
+  ArrowDownSmallIcon,
+  DeleteIcon,
+  InvisibleIcon,
+  MoreHorizontalIcon,
+  PlusIcon,
+  ToggleExpandIcon,
+  ViewIcon,
+} from '@blocksuite/icons';
+import type { Page } from '@blocksuite/store';
+import {
+  DndContext,
+  type DragEndEvent,
+  type DraggableAttributes,
+  PointerSensor,
+  useSensor,
+  useSensors,
+} from '@dnd-kit/core';
+import {
+  restrictToParentElement,
+  restrictToVerticalAxis,
+} from '@dnd-kit/modifiers';
+import { arrayMove, SortableContext, useSortable } from '@dnd-kit/sortable';
+import * as Collapsible from '@radix-ui/react-collapsible';
+import clsx from 'clsx';
+import { use } from 'foxact/use';
+import { useAtom, useAtomValue } from 'jotai';
+import type React from 'react';
+import {
+  type ChangeEventHandler,
+  type CSSProperties,
+  type MouseEvent,
+  type MouseEventHandler,
+  type PropsWithChildren,
+  Suspense,
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import { AffinePageReference } from '../reference-link';
+import { managerContext, pageInfoCollapsedAtom } from './common';
+import { getDefaultIconName, nameToIcon } from './icons-mapping';
+import {
+  type NewPropertyOption,
+  newPropertyOptions,
+  PagePropertiesManager,
+} from './page-properties-manager';
+import { propertyValueRenderers } from './property-row-values';
+import * as styles from './styles.css';
+
+type PagePropertiesSettingsPopupProps = PropsWithChildren<{
+  className?: string;
+  style?: React.CSSProperties;
+}>;
+
+const Divider = () => <div className={styles.tableHeaderDivider} />;
+
+type PropertyVisibility = PageInfoCustomProperty['visibility'];
+
+const SortableProperties = ({ children }: PropsWithChildren) => {
+  const manager = useContext(managerContext);
+  const properties = manager.getOrderedCustomProperties();
+  const readonly = manager.readonly;
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 8,
+      },
+    })
+  );
+  const onDragEnd = useCallback(
+    (event: DragEndEvent) => {
+      if (readonly) {
+        return;
+      }
+      const { active, over } = event;
+      const fromIndex = properties.findIndex(p => p.id === active.id);
+      const toIndex = properties.findIndex(p => p.id === over?.id);
+
+      if (fromIndex !== -1 && toIndex !== -1 && fromIndex !== toIndex) {
+        const newOrdered = arrayMove(properties, fromIndex, toIndex);
```

Comment:
It should be fine to change the `order` property only. If we store values in an array and move items in the array, it may cause some unexpected issues since the items at the same index may change completely.

Reply:
fixed in #5809